ts/src/renderer/phaser/classes/PhaserProjectile.ts

```diff
@@ -21,31 +24,32 @@ class PhaserProjectile extends Phaser.GameObjects.Container {
 		scene.add.existing(this);
 		scene.events.on('update', this.update, this);
 
+		this.transformListener = projectile.on('transform', (info) => {
+			this.x = info.x;
+			this.y = info.y;
+			this.sprite.rotation = info.rotation;
+		});
+
+		this.scaleListener = projectile.on('scale', (info) => {
+			console.log('scale listener', info);
+			this.sprite.setDisplaySize(info.x, info.y);
+		});
+
 		this.playAnimationListener =
 			projectile.on('play-animation', (animationId: string) => {
 				console.log('PhaserProjectile play-animation', `${key}/${animationId}`);  // TODO remove
 				sprite.play(`${key}/${animationId}`);
 			});
-	}
-
-	update (/*time: number, delta: number*/): void {
-
-		const projectile = this.projectile;
 
-		if (!projectile._alive) {
+		this.destroyListener = projectile.on('destroy', () => {
 			projectile.off('play-animation', this.playAnimationListener);
```

Comment:
should we remove the other 3 event listeners in the same manner?

Reply:
most definitely, good catch

Reply:
yes, for the `update` listener

Reply:
for destroying these Phaser entities, would it be better to make this a method of PhaserEntity when we introduce it, and just generalize the process by iterating the Entity._eventListeners array and calling `.off()` on them all?

Reply:
obviously that would not be part of this PR

Reply:
yes, good idea, now when we are working on phaser projectile we will gather ideas for phaserEntity PR